### DISCUSSION TITLE
Add "retry" option for refreshes when they fail

### DISCUSSION THREAD

**Discussion by edhans** [TabularEditor/TabularEditor3#1367] [Ideas]
The use case is large data in Dataflows where Incremental Refresh is used. The refresh of a partition can fail with a "DirectQueryUnexpectedEndOfStream" error, which is I suppose where the SQL engine just dies due to capacity overload.

When hydrating large models with many partitions with Te/3, if I refresh partitions A, B, C and D, and if B fails for any reason, B is empty and C & D don't run at all.

I'd like an option to allow up to 3 "retries" (configurable in Preferences) so when B fails, TE/3 will report it, but simply restart it and continue on.

It would be for any refresh issues, I just bring up the real world use case I experience.

**Comment by mlonsk** [comment]
Hi @edhans 
Thank you for the suggestion. I can definitely see the use case, though I believe it to be a slight edge case for most users. 

Could a workaround be to start the refresh of partitions A, B, C, and D as separate refreshes? I believe that starting them separately would allow them to run independently of each other.
